pkg/service/service.go

```diff
@@ -203,6 +203,31 @@ type L7LBInfo struct {
 	// port number for L7 LB redirection. Can be zero if only backend sync
 	// has been requested.
 	proxyPort uint16
+
+	// (sub)set of service's frontend ports to be redirected. If empty, all frontend ports will be redirected.
+	ports []uint16
+}
+
+// isPortMatch returns true if frontend has protocol TCP and its Port is in i.ports, or if i.ports
+// is empty.
+// 'ports' is typically short for no point optimizing the search.
+func (i *L7LBInfo) isPortMatch(fe *lb.L4Addr) bool {
+	// L7 LB redirect is only supported for TCP frontends
+	if fe.Protocol != lb.TCP {
+		return false
+	}
```

Comment:
Can you please break this part out into a separate patch? Also, let's call this `isProtoAndPortMatch` or similar?

Reply:
Done.